app/main.py

```diff
@@ -1,3 +1,68 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance | int | float") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        raise TypeError("Operand must be type 'Distance','int' or 'float'")
+
+    def __iadd__(self, other: "Distance | int | float") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError("Operand must be type 'Distance','int' or 'float'")
+        return self
+
+    def __mul__(self, number: float) -> "Distance":
+        return Distance(self.km * number)
+
+    def __truediv__(self, number: float) -> "Distance":
+        if number != 0:
+            return Distance(round(self.km / number, 2))
+        raise ZeroDivisionError("Division by zero is not allowed")
```

Comment:
You might want to add a check here to ensure that the 'number' argument is a non-zero number (either an integer or a float). If it's not, you could raise a TypeError or a ValueError with a helpful error message. This will make your class more robust and easier to use correctly.